overlay.nix

```diff
@@ -1,30 +1,39 @@
+{ sources }:
 self: super:
-
 with super.lib;
-
 let
-  nodePackages = self.callPackage ./pkgs/node-packages {
+  privateNodePackages = self.callPackage ./pkgs/node-packages {
     nodejs = super."nodejs-14_x";
   };
 
   # We have our own version of typescript-language-server here because the version in upstream nixpkgs
   # has a bug which is causing issues for code intelligence in node repls.
   # Copied from nixpkgs:
   # https://cs.github.com/NixOS/nixpkgs/blob/529ce4161a07f0b5a4a6d6cc4339d50f7cec77b5/pkgs/development/node-packages/default.nix#L474-L480
-  typescript-language-server = nodePackages."typescript-language-server-0.9.6".override {
+  typescript-language-server = privateNodePackages."typescript-language-server-0.9.6".override {
       nativeBuildInputs = [ self.makeWrapper ];
       postInstall = ''
         wrapProgram "$out/bin/typescript-language-server" \
           --suffix PATH : ${self.lib.makeBinPath [ self.nodePackages.typescript ]} \
           --add-flags "--tsserver-path ${self.nodePackages.typescript}/lib/node_modules/typescript/lib/"
       '';
     };
+
+  prybar = import sources.prybar {};
+
 in
-{
+rec {
```

Comment:
I think you can avoid adding this `rec` if you instead use
```
inherit (self) python310Full;
```
in the replitPackages set.

I like to avoid using rec, since it can be confusing to debug.

https://nix.dev/anti-patterns/language#rec-expression

Though, perhaps the overlay fixpoint is more confusing to debug?

Reply:
lol. The linter told me to use rec. I'd love to not use it.

Reply:
Which linter do you use?

Reply:
nix-linter. I am trying your trick.